core/src/main/kotlin/com/strumenta/kolasu/ids/SourceIdProvider.kt

```diff
@@ -24,15 +25,15 @@ abstract class AbstractSourceIdProvider : SourceIdProvider {
 
 class SimpleSourceIdProvider(var acceptNullSource: Boolean = false) : AbstractSourceIdProvider() {
```

Comment:
What's the purpose of this class? In the case of the EGL parser, I had problems (now solved) because it wasn't setting the source of the AST. Also see https://github.com/Strumenta/kolasu/issues/291

Reply:
For nodes we calculate their IDs besides on two strategies:
- "Semantic Node IDs" for nodes that represent something with a proper identity, typically something you can refer to
- "Positional Node IDs" for all other nodes. Now when calculating the Positional Node ID of a root AST node we use an id derived from its source. This class is the one that given a source gives us back an ID we can use to produce the Node ID of the root of an AST having that source

Eventually I think that Sources should produce that ID on their own and we could remove this class